src/release-specification.ts

```diff
@@ -301,6 +299,58 @@ export async function validateReleaseSpecification(
           });
         }
       }
+
+      // Check to compel users to release new versions of dependencies alongside their dependents
+      if (
+        pkg &&
+        versionSpecifierOrDirective &&
+        (hasProperty(IncrementableVersionParts, versionSpecifierOrDirective) ||
+          isValidSemver(versionSpecifierOrDirective))
+      ) {
+        const missingDependencies = Object.keys({
+          ...(pkg.unvalidatedManifest.dependencies || {}),
+          ...(pkg.unvalidatedManifest.peerDependencies || {}),
+        }).filter((dependency) => {
+          const dependencyVersionSpecifierOrDirective =
+            unvalidatedReleaseSpecification.packages[dependency];
+
+          return (
+            dependencyVersionSpecifierOrDirective !== SKIP_PACKAGE_DIRECTIVE &&
+            dependencyVersionSpecifierOrDirective !==
+              INTENTIONALLY_SKIP_PACKAGE_DIRECTIVE &&
+            !hasProperty(
+              IncrementableVersionParts,
+              dependencyVersionSpecifierOrDirective,
+            ) &&
+            !isValidSemver(dependencyVersionSpecifierOrDirective)
+          );
+        });
+
+        if (missingDependencies.length > 0) {
+          errors.push({
+            message: [
+              `The following packages, which uses a released package ${packageName}, are missing.`,
+              missingDependencies
+                .map((dependency) => `  - ${dependency}`)
+                .join('\n'),
+              " Consider including them in the release spec so that they won't break in production.",
+              `  If you are ABSOLUTELY SURE that this won't occur, however, and want to postpone the release of a package, then list it with a directive of "intentionally-skip". For example:`,
```

Comment:
Nit: I've attempted to re-word this to make the risks more explicit:
```suggestion
              ` These packages may have changes that '${packageName}' relies upon. Consider including them in the release spec.`,
              `  If you are ABSOLUTELY SURE that these packages are safe to omit, however, and want to postpone the release of a package, then list it with a directive of "intentionally-skip". For example:`,
```

Reply:
I think that this change makes sense. We'll probably want to be consistent about whether we use "depends on" or "relies on", but this is fine for now — I can give this PR another pass.